server/app/views/AzureFileUploadViewStrategy.java

```diff
@@ -112,6 +112,10 @@ public Tag renderFileUploadBlockSubmitForms(
 
   @Override
   protected Optional<ContainerTag> maybeRenderSkipOrDeleteButton(Params params) {
+    if (hasAtLeastOneRequiredQuestion(params)) {
+      // If the file question is required, skip or delete is not allowed.
```

Comment:
If both the functions do the same, should we just move this to the FileUploadViewStrategy.java?

Reply:
there's definitely opportunity for deeper refactoring here, but this quick fix will resolve the bug in prod

Reply:
I definitely agree. There's a lot of UI building code here that would benefit from deduplication. For now, I'm going to proceed with the quick fix.